src/structlog/_base.py

```diff
@@ -52,17 +36,22 @@ class BoundLoggerBase:
         See also `custom-wrappers`.
     """
 
-    def __init__(self, logger, processors, context):
+    def __init__(
+        self,
+        logger: WrappedLogger,
+        processors: Sequence[Processor],
+        context: Context,
+    ):
         self._logger = logger
         self._processors = processors
         self._context = context
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "<{}(context={!r}, processors={!r})>".format(
             self.__class__.__name__, self._context, self._processors
         )
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
```

Comment:
usually for `__eq__` I've seen `other: object` instead of `other: Any` (this makes it an opaque object instead of a squishy one) -- that is it's an error to try and access anything on it instead of allowing everything

though, looking at the body of this function it's maybe better to leave it as `Any` for now and adjust the code to be more type-friendly later